app/src/main/java/de/danoeh/antennapod/dialog/FeedRefreshIntervalDialog.java

```diff
@@ -0,0 +1,161 @@
+package de.danoeh.antennapod.dialog;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.res.Resources;
+import android.os.Build;
+import android.text.format.DateFormat;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.widget.ArrayAdapter;
+import android.widget.RadioButton;
+import android.widget.RadioGroup;
+import android.widget.Spinner;
+import android.widget.TimePicker;
+
+import androidx.appcompat.app.AlertDialog;
+
+import org.apache.commons.lang3.ArrayUtils;
+
+import java.util.concurrent.TimeUnit;
+
+import de.danoeh.antennapod.R;
+import de.danoeh.antennapod.core.preferences.UserPreferences;
+
+public class FeedRefreshIntervalDialog {
+
+    public interface OnUpdateCallback {
+        void handleUpdate();
+    }
+
+    public static void show(Activity activity, OnUpdateCallback onUpdateCallback) {
+        final Context context = activity;
+        AlertDialog.Builder builder = new AlertDialog.Builder(context);
+        builder.setTitle(R.string.pref_autoUpdateIntervallOrTime_titleAct);
+        builder.setMessage(R.string.pref_autoUpdateIntervallOrTime_sumAct);
+        LayoutInflater inflater = activity.getLayoutInflater();
+
+        View view = inflater.inflate(R.layout.update_refresh_time_dialoxg, null);
+        builder.setView(view);
+
+        final String[] intervalSpinnerValues = context.getResources().getStringArray(R.array.update_intervall_values);
+        final String[] entries = appendHourUnit(activity, intervalSpinnerValues);
+        long currInterval = UserPreferences.getUpdateInterval();
+        int checkedItem = -1;
+
+        if (currInterval > 0) {
+            String currIntervalStr = String.valueOf(TimeUnit.MILLISECONDS.toHours(currInterval));
+            checkedItem = ArrayUtils.indexOf(intervalSpinnerValues, currIntervalStr);
+        }
+
+        ArrayAdapter<String> spinnerArrayAdapter = new ArrayAdapter<String>(context,
+                android.R.layout.simple_spinner_item, entries);
+        spinnerArrayAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+
+        int hourOfDay = 7;
+        int minute = 0;
+        int[] updateTime = UserPreferences.getUpdateTimeOfDay();
+        if (updateTime.length == 2) {
+            hourOfDay = updateTime[0];
+            minute = updateTime[1];
+        }
+
+        int finalCheckedItem = checkedItem;
+        int finalHourOfDay = hourOfDay;
+        int finalMinute = minute;
+
+        Spinner spinner = (Spinner) view.findViewById(R.id.spinner);
+        RadioButton radioButtonInterval = (RadioButton) view.findViewById(R.id.intervalRadioButton);
+        RadioButton radioButtonTime = (RadioButton) view.findViewById(R.id.timeRadioButton);
+        RadioButton radioButtonDisable = (RadioButton) view.findViewById(R.id.disableButton);
+        RadioGroup radioGroupOne = (RadioGroup) view.findViewById(R.id.genderRadioGroup);
```

Comment:
The first radio button is always checked by default, even if the setting is actually set to `Disabled`. Please change the visibility here, depending on the value of the preference. Maybe there could even be a method that does this, so that it can be reused in `onCheckedChanged`.